src/GraphQL.DataLoader/DataLoaderContextAccessor.cs

```diff
@@ -6,9 +6,10 @@ public class DataLoaderContextAccessor : IDataLoaderContextAccessor
     private static readonly AsyncLocal<DataLoaderContext?> _current = new();
 
     /// <inheritdoc/>
-    public DataLoaderContext? Context
+    [AllowNull]
+    public DataLoaderContext Context
     {
-        get => _current.Value;
+        get => _current.Value!;
```

Comment:
I'm a little unsure about this. On the one hand, using `!` here will result in the least amount of code when the data loader feature is used correctly. But when used incorrectly, it returns null rather than having a helpful exception message. I have a hard time penalizing the correct case for this, though.

An alternative would be to use a throw helper to throw when it is null. The property could still be inlined, but there's still a conditional check...

Reply:
Ah, another option I just thought of: `Debug.Assert`. That will help developers ensure it is setup correctly while not penalizing production use.

Reply:
Debug.Assert are for internal testing; the compiler removes these from the compiled nuget packages. So developers won’t get any benefit from Debug.Assert.

Reply:
🤦‍♂️ of course. I'll remove that. Do you think I should add a throw helper or leave it without one?

Reply:
I probably would. I can’t think of a reason you’d need to know the value was null. However I’d consider it a breaking change and push the change to v8

Reply:
Adding the throw helper assists the developer in knowing how to fix the issue which is a great benefit.

Reply:
Great, I'll change the base to the develop branch and add the throw helper.